towerlib/towerlib.py

```diff
@@ -1399,21 +1399,27 @@ def create_credential_in_organization(self,  # pylint: disable=too-many-argument
 
         Raises:
             InvalidOrganization: The organization provided as argument does not exist.
-            InvalidUser: The user provided as argument does not exist.
-            InvalidTeam: The team provided as argument does not exist.
             InvalidCredentialType: The credential type provided as argument does not exist.
             InvalidVariables: The inputs provided as argument is not valid json.
 
         """
+        team_id = None
+        user_id = None
         organization_ = self.get_organization_by_name(organization)
         if not organization_:
             raise InvalidOrganization(organization)
-        user_ = self.get_user_by_username(user)
-        if not user_:
-            raise InvalidUser(user)
-        team_ = organization_.get_team_by_name(team)
-        if not team_:
-            raise InvalidTeam(team)
+        if user:
+            user_ = self.get_user_by_username(user)
+            if not user_:
+                raise InvalidUser(user)
+            else:
+                user_id = user_.id
+        if team:
+            team_ = organization_.get_team_by_name(team)
+            if not team_:
+                raise InvalidTeam(team)
+            else:
```

Comment:
The else is not needed here since if the ```if``` triggers you will get an exception. Just let the execution flow through.